### DISCUSSION TITLE
Failed to install plugin 'js' version 4.0.0. on Windows 10

### DISCUSSION THREAD

**Discussion by retojs** [getgauge/gauge#2447] [General]
I've installed gauge 1.5.6. but installing the js plugin fails: 

PS C:\Windows\system32> gauge install js -l debug
Gathering metadata for js
Downloading https://downloads.gauge.org/plugin/js?l=&p=html-report,screenshot&o=windows&a=amd64
Downloading https://github.com/getgauge/gauge-js/releases/download/v4.0.0/gauge-js-4.0.0.zip
.......................
Running plugin hook command => [npm install --omit=dev --silent]
Failed to install plugin 'js' version 4.0.0.
Reason: exit status 1

What can I do?

**Comment by chadlwilson** [comment]
Can you look at https://github.com/getgauge/gauge-js/issues/616 to find some ways to try and get additional information about why the plugin install is failing and your specific environment?

**Reply by retojs** [reply]
Are you referring to this comment of yours in https://github.com/getgauge/gauge-js/issues/616 ? 

> I believe there should be a debug log within the cache folder (npm config get cache), either npm-debug.log or one of the log files in _logs. Can you look for that and see what went wrong?

I found those logs and I can see output from my previous npm commands in there, but the _gauge install js_ command does not produce any log output.

**Reply by chadlwilson** [reply]
There are a number of suggestions there, including trying the npm install manually from the plugin directory to see what the error message is.

**Reply by retojs** [reply]
When I unzip https://github.com/getgauge/gauge-js/releases/download/v4.0.0/gauge-js-4.0.0.zip into gauge/plugins/js/4.0.0 and execute _npm install --omit=dev_ inside that folder im getting the error:
```
npm ERR! code EEXIST
npm ERR! path ~\AppData\Roaming\gauge\plugins\js\4.0.0\node_modules\.bin\esgenerate
npm ERR! Refusing to delete ~\AppData\Roaming\gauge\plugins\js\4.0.0\node_modules\.bin\esgenerate: is outside ~\AppData\Roaming\gauge\plugins\js\4.0.0\node_modules\escodegen and not a link

```
With option **--force** the plugin can successfully be installed.

**Reply by chadlwilson** [reply]
This looks like a pretty common npm problem on windows from what I can see. No idea the root cause though, although I suspect the sadly usual Windows filesystem problems, especially as it relates to either symbolic links (which windows doesn’t really support) or permissions.

The 3.0.0+ releases use different versions of escodegen so if there was a problem with your install or the cached version of escodegen 2.1.0 in npm-cache that might explain why installing the older plugin version was ok.

Other than forcing the delete, sounds like people normally clean their npm cache and/or node_modules.

is your issue fixed now?

**Reply by retojs** [reply]
For me the issue is resolved. Thanks for your support.

**Comment by retojs** [comment]
I've got the same problem with gauge-js v3.0.1 and v3.0.0. But I could install version v2.4.0.

**Reply by chadlwilson** [reply]
Could be something to do with Node or NPM version available on your path, hence asking for that information.
